lib/mjsonwp/routes.js

```diff
@@ -455,7 +455,56 @@ for (let v of _.values(METHOD_MAP)) {
   }
 }
 
+const RE_ESCAPE = /[\-\[\]{}()+?.,\\\^$|#\s]/g;
+const RE_PARAM = /([:*])(\w+)/g;
+
+class Route {
+  constructor (route) {
+    this.paramNames = [];
+    this.routeRegexp = route;
+
+    this.routeRegexp = this.routeRegexp.replace(RE_ESCAPE, "\\$&");
+    this.routeRegexp = this.routeRegexp.replace(RE_PARAM, (_, mode, name) => {
+      this.paramNames.push(name);
+      return mode === ":" ? "([^/]*)" : "(.*)";
+    });
+    this.routeRegexp = new RegExp("^" + this.routeRegexp + "$");
```

Comment:
Also, until this point, using `this.routeRegexp` seems odd. I'd rather use a local variable to deal with the build up. Something like 
```js
let reStr = route.replace(RE_ESCAPE, "\\$&");
reStr = reStr.replace(RE_PARAM, (_, mode, name) => {
  this.paramNames.push(name);
  return mode === ":" ? "([^/]*)" : "(.*)";
});
this.routeRegexp = new RegExp(`^${reStr}$`);
```